File: android/resize_assets.py
"""
Resize JPM icon and splash screen to all required Android densities
and copy them into the correct res/ folders.
Run from: web/android/
"""

import sys, shutil
from pathlib import Path
from PIL import Image

BASE = Path(__file__).parent
RES  = BASE / "app" / "src" / "main" / "res"

ICON_SRC   = Path(r"C:\Users\Joshan\.gemini\antigravity\brain\f3061532-f698-4a28-a008-25856d571be9\jpm_icon_1773016920540.png")
SPLASH_SRC = Path(r"C:\Users\Joshan\.gemini\antigravity\brain\f3061532-f698-4a28-a008-25856d571be9\jpm_splash_1773016934605.png")

# Icon sizes per density: (folder, size_px)
ICON_DENSITIES = [
    ("mipmap-mdpi",    48),
    ("mipmap-hdpi",    72),
    ("mipmap-xhdpi",   96),
    ("mipmap-xxhdpi",  144),
    ("mipmap-xxxhdpi", 192),
]

# Splash sizes per density: (folder, width, height)
SPLASH_DENSITIES = [
    ("drawable",              1080, 1920),
    ("drawable-port-mdpi",    320,  480),
    ("drawable-port-hdpi",    480,  800),
    ("drawable-port-xhdpi",   720,  1280),
    ("drawable-port-xxhdpi",  1080, 1920),
    ("drawable-port-xxxhdpi", 1440, 2560),
    ("drawable-land-mdpi",    480,  320),
    ("drawable-land-hdpi",    800,  480),
    ("drawable-land-xhdpi",   1280, 720),
    ("drawable-land-xxhdpi",  1920, 1080),
    ("drawable-land-xxxhdpi", 2560, 1440),
]

def resize_icon(src: Path, dst: Path, size: int):
    img = Image.open(src).convert("RGBA")
    img = img.resize((size, size), Image.LANCZOS)
    dst.parent.mkdir(parents=True, exist_ok=True)
    img.save(dst, "PNG")
    print(f"  icon → {dst.relative_to(BASE)} ({size}x{size})")

def resize_splash(src: Path, dst: Path, w: int, h: int):
    img = Image.open(src).convert("RGBA")
    # Fit into target dimensions keeping aspect ratio, pad with white
    img.thumbnail((w, h), Image.LANCZOS)
    canvas = Image.new("RGBA", (w, h), (255, 255, 255, 255))
    canvas.paste(img, ((w - img.width)//2, (h - img.height)//2))
    dst.parent.mkdir(parents=True, exist_ok=True)
    canvas.save(dst, "PNG")
    print(f"  splash → {dst.relative_to(BASE)} ({w}x{h})")

print("=== Resizing icons ===")
for folder, size in ICON_DENSITIES:
    for name in ("ic_launcher.png", "ic_launcher_round.png", "ic_launcher_foreground.png"):
        resize_icon(ICON_SRC, RES / folder / name, size)

print("\n=== Resizing splash screens ===")
for folder, w, h in SPLASH_DENSITIES:
    resize_splash(SPLASH_SRC, RES / folder / "splash.png", w, h)

print("\n✅  Done! All assets replaced.")

File: android/resize_assets.js
const sharp = require('sharp');
const path = require('path');
const fs = require('fs');

const ICON_SRC   = String.raw`C:\Users\Joshan\.gemini\antigravity\brain\f3061532-f698-4a28-a008-25856d571be9\jpm_icon_1773016920540.png`;
const SPLASH_SRC = String.raw`C:\Users\Joshan\.gemini\antigravity\brain\f3061532-f698-4a28-a008-25856d571be9\jp_splash_1773017830589.png`;
const RES = path.join(__dirname, 'app', 'src', 'main', 'res');

const ICON_DENSITIES = [
  ['mipmap-mdpi',    48],
  ['mipmap-hdpi',    72],
  ['mipmap-xhdpi',   96],
  ['mipmap-xxhdpi',  144],
  ['mipmap-xxxhdpi', 192],
];

const SPLASH_DENSITIES = [
  ['drawable',              1080, 1920],
  ['drawable-port-mdpi',    320,  480 ],
  ['drawable-port-hdpi',    480,  800 ],
  ['drawable-port-xhdpi',   720,  1280],
  ['drawable-port-xxhdpi',  1080, 1920],
  ['drawable-port-xxxhdpi', 1440, 2560],
  ['drawable-land-mdpi',    480,  320 ],
  ['drawable-land-hdpi',    800,  480 ],
  ['drawable-land-xhdpi',   1280, 720 ],
  ['drawable-land-xxhdpi',  1920, 1080],
  ['drawable-land-xxxhdpi', 2560, 1440],
];

async function run() {
  // Icons
  for (const [folder, size] of ICON_DENSITIES) {
    const dir = path.join(RES, folder);
    fs.mkdirSync(dir, { recursive: true });
    for (const name of ['ic_launcher.png', 'ic_launcher_round.png', 'ic_launcher_foreground.png']) {
      const dest = path.join(dir, name);
      await sharp(ICON_SRC).resize(size, size).png().toFile(dest);
      console.log(`icon → ${folder}/${name} (${size}x${size})`);
    }
  }

  // Splash screens — pad with white to fill target resolution
  for (const [folder, w, h] of SPLASH_DENSITIES) {
    const dir = path.join(RES, folder);
    fs.mkdirSync(dir, { recursive: true });
    const dest = path.join(dir, 'splash.png');
    await sharp({
      create: {
        width: w,
        height: h,
        channels: 4,
        background: { r: 10, g: 10, b: 10, alpha: 1 }
      }
    })
      .png()
      .toFile(dest);
    console.log(`splash (blank) → ${folder}/splash.png (${w}x${h})`);
  }

  console.log('\n✅  All Android assets replaced!');
}

run().catch(e => { console.error(e); process.exit(1); });
